### PR TITLE
Further improvements for Tuple's documentation

### PR DESCRIPTION
**Pull request by MetaLang**
- Formatting and wording fixes
- Fix documented unittest for expand
- Add Params and Returns where appropriate
- Document toHash

### CODE REVIEW

File: std/typecons.d [hunk 1]
```diff
@@ -444,13 +455,29 @@ template Tuple(Specs...)
         }
 
         /**
-         * Use $(D t.expand) for a tuple $(D t) to expand it into its
-         * components. The result of $(D expand) acts as if the tuple components
+         * Use $(D t.expand) for a `Tuple` $(D t) to expand it into its
+         * components. The result of $(D expand) acts as if the `Tuple's` components
```
Comment (Hackerpilot) [inline std/typecons.d]: \`Tuple\`'s

File: std/typecons.d [hunk 2]
```diff
@@ -568,6 +594,13 @@ template Tuple(Specs...)
          *   $(LI For each value `v1` on the left-hand side and each value 
          *        `v2` on the right-hand side, the expression `v1 == v2` is 
          *        true.))
+         *
+         * Params:
+         *     rhs = The `Tuple` to compare against. It must meeting the criteria
+         *           for comparison between `Tuples`.
```
Comment (Hackerpilot) [inline std/typecons.d]: \`Tuple\`s

File: std/typecons.d [hunk 3]
```diff
@@ -568,6 +594,13 @@ template Tuple(Specs...)
...
+         *
+         * Returns:
+         *     true if both `Tuples` are equal, otherwise false.
```
Comment (Hackerpilot) [inline std/typecons.d]: Same here.

File: std/typecons.d (hunk 1 above)
Comment (MetaLang) [inline std/typecons.d]: Looks incredibly ugly but I guess I can change them.

**Comment by MetaLang** [comment]
I have fixed the placement of s and 's following `Tuple`.

File: std/typecons.d (hunk 1 above)
Comment (Hackerpilot) [inline std/typecons.d]: I agree, but I also think it's important that anything inside of the \`\` be actual types or code.

**Comment by MetaLang** [comment]
Ping

**Comment by Hackerpilot** [comment]
I don't actually have the ability to merge anything.

File: std/typecons.d [hunk 4]
```diff
@@ -568,6 +594,13 @@ template Tuple(Specs...)
          *   $(LI For each value `v1` on the left-hand side and each value 
          *        `v2` on the right-hand side, the expression `v1 == v2` is 
          *        true.))
+         *
+         * Params:
+         *     rhs = The `Tuple` to compare against. It must meeting the criteria
+         *           for comparison between `Tuple`s.
+         *
+         * Returns:
+         *     true if both `Tuples` are equal, otherwise false.
```
Comment (burner) [inline std/typecons.d]: you used `Tuple`s above please make it consistent

also: true if the elements of the tuples are equal

File: std/typecons.d [hunk 5]
```diff
@@ -676,7 +717,16 @@ template Tuple(Specs...)
         }
 
         /**
-         * Takes a slice of the `Tuple`.
+         * Takes a slice of this `Tuple`.
+         *
+         * Params:
+         *     from = A `size_t` designating the starting position to slice from.
+         *     to = A `size_t` designating the ending position.
```
Comment (burner) [inline std/typecons.d]: of the slice?

File: std/typecons.d [hunk 6]
```diff
@@ -696,6 +746,12 @@ template Tuple(Specs...)
             assert(s[0] == "abc" && s[1] == 4.5);
         }
 
+        /**
+            Creates a hash of this `Tuple`.
+            
+            Returns:
+                A unique `size_t` representing the hash of this `Tuple`.
```
Comment (burner) [inline std/typecons.d]: IMO unqiue is wrong here. e.g. there can not be a unique hash of type size_t for Tuple!(size_t,size_t)

File: std/typecons.d (hunk 4 above)
Comment (MetaLang) [inline std/typecons.d]: Isn't it the same thing? Two Tuples can only be equal if their elements are equal.

**Comment by MetaLang** [comment]
I have fixed the issues that Robert pointed out.

File: std/typecons.d (hunk 4 above)
Comment (burner) [inline std/typecons.d]: likely not if a language has "is" "opEquals" ...

Comment (MetaLang) [inline std/typecons.d]: But this is specifically how Tuple comparison is implemented; if each of its members are equal.

Comment (burner) [inline std/typecons.d]: my bad

**Comment by burner** [comment]
Auto-merge toggled on

**Pull request merged**